components/brave_new_tab_ui/apiEventsToStore.ts

```diff
@@ -32,6 +33,11 @@ function onRewardsToggled (prefData: preferencesAPI.Preferences): void {
   }
 }
 
+async function onMostVisitedInfoChanged (topSites: topSitesAPI.MostVisitedInfoChanged) {
+  getActions().tilesUpdated(topSites.tiles)
+  getActions().topSitesStateUpdated(topSites.visible, topSites.custom_links_enabled)
```

Comment:
(at this point this can be a follow up): definitely make this dispatch a single action since both these 2 dispatches modify state, and we have all the data here to just send it in a single dispatch and reduce cycles.